### DISCUSSION TITLE
Where are you getting the player data from? Where do you get the player ranks from? From a developer POV

### DISCUSSION THREAD

**Discussion by Andrew-Bernhardt** [TeKrop/overfast-api#37] [Q&A] [answered]
What call are you making to get the players rank? Thanks!

**Comment by TeKrop** [comment, marked as answer]
Hello ! For players datas, I'm using two endpoints : 
- **Players search** : `https://overwatch.blizzard.com/search/account-by-name/{name}`. It's the endpoint called from this page : 
`https://overwatch.blizzard.com/en-us/search/?q={name}`
- **Players stats** : `https://overwatch.blizzard.com/career/{player_battletag}`. Ex : `https://overwatch.blizzard.com/career/TeKrop-2217`

**Reply by Andrew-Bernhardt** [reply]
Okay, thank you. So if I wanted to get the player data, i.e. rank for their tank role, how do I go about getting that? Do I make a get request of that page? Or are you scraping the data off that page via a webscraper? Trying to learn as a developer, thank you.

**Reply by TeKrop** [reply]
As you guessed, I'm scraping the data off that page, by using Beautiful Soup (https://beautiful-soup-4.readthedocs.io/en/latest/). If you want to retrieve the data, I can give you two ways to do it : 
- Either use Beautiful Soup like me in order to learn how to use a scraper with python. If you're using another language, you'll certainly find another popular scraper for it.
- Either use my API and handle data exposed in JSON format (quite popular data format), by calling the player summary endpoint (`https://overfast-api.tekrop.fr/players/{player_id}/summary`). The `player_id` is the battletag of the player, with the `#` replaced by a `-`. You can find the associated documentation here, with an example on the right : https://overfast-api.tekrop.fr/#tag/Players/operation/get_player_summary_players__player_id__summary_get 

I hope I answered all your questions, good luck with your learning 😄
